complex_functions/SearchProblem.py

```diff
@@ -0,0 +1,79 @@
+class SearchProblem:
```

Comment:
We have had the convention to add the link to report on each file that we took from the repo, see the other complex functions as an example